UnitTests/UnitTests.UWP/UI/Collection/Test_IncrementalLoadingCollection.cs

```diff
@@ -28,76 +25,83 @@ public class Test_IncrementalLoadingCollection
                 DataSource<int>.ThrowException, DataSource<int>.PassThrough,
             };
 
-        private static readonly int[] AllData
-            = Enumerable.Range(0, Pages * PageSize).ToArray();
-
-        [DataRow]
-        [DataRow(2500, 1000, 1000, 1000, 1000)]
         [TestMethod]
-        public async Task Requests(params int[] pageDelays)
+        [DataRow(20, 5)]
+        [DataRow(5, 5)]
+        [DataRow(20, 5, 2500, 1000, 1000, 1000, 1000)]
+        public async Task Requests(int pageSize, int pages, params int[] pageDelays)
         {
-            var source = new DataSource<int>(AllData, pageDelays.Select(DataSource<int>.MakeDelayOp));
-            var collection = new IncrementalLoadingCollection<DataSource<int>, int>(source, PageSize);
+            var allData = Enumerable.Range(0, pages * pageSize).ToArray();
+            var source = new DataSource<int>(allData, pageDelays.Select(DataSource<int>.MakeDelayOp));
+            var collection = new IncrementalLoadingCollection<DataSource<int>, int>(source, pageSize);
 
-            for (int pageNum = 1; pageNum <= Pages; pageNum++)
+            for (var pageNum = 1; pageNum <= pages; pageNum++)
             {
                 await collection.LoadMoreItemsAsync(0);
-                CollectionAssert.AreEqual(Enumerable.Range(0, PageSize * pageNum).ToArray(), collection);
+                CollectionAssert.AreEqual(Enumerable.Range(0, pageSize * pageNum).ToArray(), collection);
             }
         }
 
-        [DataRow]
-        [DataRow(2500, 1000, 1000, 1000, 1000)]
         [TestMethod]
-        public async Task RequestsAsync(params int[] pageDelays)
+        [DataRow(20, 5)]
+        [DataRow(5, 5)]
+        [DataRow(20, 5, 2500, 1000, 1000, 1000, 1000)]
+        public async Task RequestsAsync(int pageSize, int pages, params int[] pageDelays)
         {
-            var source = new DataSource<int>(AllData, pageDelays.Select(DataSource<int>.MakeDelayOp));
-            var collection = new IncrementalLoadingCollection<DataSource<int>, int>(source, PageSize);
+            var allData = Enumerable.Range(0, pages * pageSize).ToArray();
+            var source = new DataSource<int>(allData, pageDelays.Select(DataSource<int>.MakeDelayOp));
+            var collection = new IncrementalLoadingCollection<DataSource<int>, int>(source, pageSize);
 
             var requests = new List<Task>();
 
-            for (int pageNum = 1; pageNum <= Pages; pageNum++)
+            for (var pageNum = 1; pageNum <= pages; pageNum++)
             {
                 requests.Add(collection.LoadMoreItemsAsync(0).AsTask()
                     .ContinueWith(t => Assert.IsTrue(t.IsCompletedSuccessfully)));
```

Comment:
Should we check the interim results for pageSize here as well as we're passing that into the test now?

Reply:
I am not sure its that easy since these calls are happening asynchronously and we would need to test ageist `collection` which will be changing as these tasks complete.